lld/test/ELF/debug-names-bad-name-count.s

```diff
@@ -1,22 +1,29 @@
 # REQUIRES: x86
 # RUN: llvm-mc -filetype=obj -triple=x86_64 %s -o %t1.o
-	
+
 # RUN: not ld.lld --debug-names %t1.o -o /dev/null 2>&1 \
-# RUN:   FileCheck -DFILE=%t1.o --implicit-check-not=error: %s
-	
+# RUN:   | FileCheck -DFILE=%t1.o --implicit-check-not=error: %s
+
 # CHECK: error: [[FILE]]:(.debug_names): index entry is out of bounds
 # CHECK: error: [[FILE]]:(.debug_names): index entry is out of bounds
 # CHECK: error: [[FILE]]:(.debug_names): index entry is out of bounds
 # CHECK: error: [[FILE]]:(.debug_names): index entry is out of bounds
 
-# This file was generated by first compiling main.cpp:
-# clang++ -g -O0 -S -gpubnames -fdebug-compilation-dir='debug-names-test' \
-#      main.cpp
+# RUN: ld.lld --noinhibit-exec --debug-names %t1.o -o /dev/null 2>&1 \
+# RUN:   | FileCheck -DFILE=%t1.o %s --check-prefix=WARN
+
+# WARN: warning: [[FILE]]:(.debug_names): index entry is out of bounds
```

Comment:
This shows the same error message as `lld/test/ELF/debug-names-bad-die-idx-sizes.s` - is it testing a different error path (if it is, should the error message be more specific to distinguish these cases?) - or is it the same code path/error, in which case maybe not worth testing?

Reply:
They test different types of errors in the input, although both of those input errors are caught by the same code path in LLD.  @MaskRay do you think we need to keep both tests, or not?

Reply:
If they are different errors, having the test is fine.

If the `sed` pattern is applied (https://github.com/llvm/llvm-project/pull/86508#discussion_r1565261694), it might be clear about the different types of errors.

Reply:
That does look like it'd reduce the verbosity/repetition, which I think is good.

But in this case I think they're equivalent errors ( https://en.wikipedia.org/wiki/Equivalence_class ) - different inputs that fail in the same way. I think it'd be fine to only test the class once.